### ISSUE TITLE
Cosmological distance vectorization 

### ISSUE THREAD

**Issue by pennalima** [LSSTDESC/CCL#822]
Currently the cosmological distances can be computed passing the scale factors a1 and a2, or two lists with the same size. 
We need a third option: one single scale factor (lens) and a n-size list (sources) as inputs. 
@elisachisari

**Comment by elisachisari** [comment]
I think what needs to be done is to add a case similar to _vectorize_fn5 in this file: https://github.com/LSSTDESC/CCL/blob/master/pyccl/pyutils.py

**Comment by damonge** [comment]
I would simply insert the following code here: https://github.com/LSSTDESC/CCL/blob/46692a38afd249946ac004cec391f47fd7c34f63/pyccl/background.py#L135

```
if (np.ndim(a2) == 0) and (np.ndim(a1) != 0):
    a2 = np.full(len(a1), a2)
```

**Comment by damonge** [comment]
@pennalima @elisachisari fixed in #950 . Tagged you for a quick PR if you have time.
